File: backend/src/main/java/com/lerneon/backend/models/payload/request/EmailRequest.java
package com.lerneon.backend.models.payload.request;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class EmailRequest {
    @NotBlank(message = "Email cannot be blank")
    @Email(message = "Please provide a valid email address")
    @Size(min = 5, message = "Email is too short. Please enter at least 5 characters")
    @Size(max = 100, message = "Email is too long. Please enter no more than 100 characters")
    private String email;
}

File: client/types/payload/request/email.request.ts
import { z } from "zod/v3";

export const EmailRequest = z.object({
  email: z
    .string({
      required_error: "Email cannot be blank",
      invalid_type_error: "Email must be a string",
    })
    .email({ message: "Please provide a valid email address" })
    .min(1, { message: "Email cannot be blank" })
    .min(5, {
      message: "Email is too short. Please enter at least 5 characters",
    })
    .max(100, {
      message: "Email is too long. Please enter no more than 100 characters",
    }),
});

export type EmailRequest = z.infer<typeof EmailRequest>;
